apps/hash-graph/lib/graph/src/store/query.rs

```diff
@@ -13,6 +18,39 @@ pub trait QueryPath {
     fn expected_type(&self) -> ParameterType;
 }
 
+/// Parses a query token of the form `token(key=value)`.
+///
+/// Whitespaces are ignored and multiple parameters are supported.
+///
+/// # Errors
+///
+/// - If the token is not of the form `token`, `token()`, or `token(key=value)`
+/// - If `token` can not be deserialized into `T`
+pub fn parse_query_token<'de, T: Deserialize<'de>, E: de::Error>(
```

Comment:
Currently works, nothing major, but I see some caveats I'd like to see documented:

A) quoted strings do not currently work (what if - in the future - I have whitespace or `,` in my value?)
B) documentation does not specify that multiple key-value pairs are supported.

I would usually go with something like `nom`, `winnow`, `chumsky` as a parser combinator, to be future proof and have more readable code, but I understand that adding a new dependency for this might not be warranted.

Reply:
I would also go with an external parser for proper parsing but as we only split by `,` to detect well-defined strings and numbers this is by far the easiest solution.

Reply:
I also wouldn't pass `Deserialize` here but a `FromStr` instead.